mlir/test/lib/Dialect/SCF/TestSCFUtils.cpp

```diff
@@ -214,6 +214,7 @@ struct TestSCFPipeliningPass
     RewritePatternSet patterns(&getContext());
     mlir::scf::PipeliningOption options;
     options.getScheduleFn = getSchedule;
+    options.supportDynamicLoops = true;
```

Comment:
If there is an option for turning dynamic loop pipelining off, what are the cases where we don't want this option? Based on my read of this PR, it seems like when maxStage > numIterations, it might be worse to pipeline + predicate than to leave the loop alone, hence the option. If there are cases where we want the option off, can we add a test showing that it does what is expected?